### PR TITLE
Reduce the time to 

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

I am testing to reduce the timeouts for syncing SCC products so that the acceptance tests for PR do not take that long.

## GUI diff

No difference.

Before:

After:

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
Reducing to 3600 made no difference.

Let's be more aggressive and try with 600 seconds.

**Comment by jordimassaguerpla** [comment]
> Reducing to 3600 made no difference.
> 
> Let's be more aggressive and try with 600 seconds.

With 600, the core setup takes 40 minutes instead of 1h and 15 minutes. The "Initialize clients" phase passed after the core setup.

@srbarrios @Bischoff the question is, how can I make 600 be the timeout instead of 7200 when running this test at the PR? I assume you have 7200 for a reason and you want to keep that when running the test for master branch. Can we make it configurable somehow? What is the simplest solution here?

**Comment by srbarrios** [comment]
> > Reducing to 3600 made no difference.
> > Let's be more aggressive and try with 600 seconds.
> 
> With 600, the core setup takes 40 minutes instead of 1h and 15 minutes. The "Initialize clients" phase passed after the core setup.
> 
> @srbarrios @Bischoff the question is, how can I make 600 be the timeout instead of 7200 when running this test at the PR? I assume you have 7200 for a reason and you want to keep that when running the test for master branch. Can we make it configurable somehow? What is the simplest solution here?

This timeout waits until we can assure that the channel is fully sync, checking the existence of a file in a expected folder. But in case of a timeout, we are catching the exception and continue (see line 346), without returning an exception, this was done just in case we pass wrong MU URLs at some point (with a wrong channel that will never be found on that folder) but we must revisit this (@Bischoff fyi), and the good practice might be to don't hide this exception.

If the core setup was sped up from 75min to 40min is due to failing faster, but it doesn't mean it's stable to continue the tests as it must be that we still having pending synchronization processes.

To summarize, we can fail fast ofc, but that will speed up results (failed results) but not the tests themselves.

We can also parametrize this depending on the environment where we run it, for that, I would create a global variable as we do in this example, but specifying as value the kind of environment we run:
``` 
$long_tests_enabled = true if ENV['LONG_TESTS'] == 'true'
``` 
For instance, something like:
``` 
$type_environment = ENV['TYPE_ENVIRONMENT']
or
$kind_of_env = ENV['TYPE_ENVIRONMENT']
```

**Comment by jordimassaguerpla** [comment]
Do we need to have the channel full synced for the tests to run? Or just the metadata?

**Comment by jordimassaguerpla** [comment]
@srbarrios I just introduced the use of an environment variable as you suggested. This is to overwrite the default, so nothing should change for the master acceptance tests. This way I can set this for the acceptance tests for the PR. We can set this as experimental and give it a try for some weeks and see.

**Review by srbarrios — APPROVED** [review]

**Comment by Bischoff** [comment]
ok. `type_environment` is not really self-explanatory. What do you mean @jordimassaguerpla ? If you meant "type of environment" it would rather be `environment_type`, in correct English.

**Comment by Bischoff** [comment]
We want to get rid of `LONG_TESTS`, so it's not really one of the best examples.

I'd also like to have a common value for all environments - no env variable.

But something's deeply wrong with this PR. Hold on.

**Comment by Bischoff** [comment]
After discussing with Jordi, I asked him to simply revert @srbarrios .

We can do 3 things with reposync:
1. let it running during the rest of the test suite
2. kill it
3. wait until it finishes

1. is what we did before introducing those two functions. **we do not want that** anymore as it generated race conditions. Unfortunately, Jordi's PR returned to this situation.

If you really don't want to wait for reposyncs nor to kill them, then you should simply not call those two methods. But that's dangerous.

This needs more discussion. For now, let's revert.
